(rpt2 plugin) Error: Could not find source file: .../dist/index.d.ts

I have only one file in `src/index.ts`

```
{
  "name": "react-modern-calendar",
  "version": "0.0.1",
  "license": "MIT",
  "source": "src/index.ts",
  "main": "dist/index.js",
  "module": "dist/index.module.js",
  "unpkg": "dist/index.umd.js",
  "types": "dist/index.d.ts",
  "scripts": {
    "build": "microbundle",
    "dev": "microbundle watch"
  }
}
```

```
$ microbundle
rpt2: options error TS6053: File '/mnt/ssd/repositories/react-modern-calendar/dist/index.d.ts' not found.
(rpt2 plugin) Error: Could not find source file: '/mnt/ssd/repositories/react-modern-calendar/dist/index.d.ts'.
Error: Could not find source file: '/mnt/ssd/repositories/react-modern-calendar/dist/index.d.ts'.
    at getValidSourceFile (/mnt/ssd/repositories/react-modern-calendar/node_modules/typescript/lib/typescript.js:139148:29)
    at Object.getEmitOutput (/mnt/ssd/repositories/react-modern-calendar/node_modules/typescript/lib/typescript.js:139533:30)
    at /mnt/ssd/repositories/react-modern-calendar/node_modules/rollup-plugin-typescript2/dist/rollup-plugin-typescript2.cjs.js:27438:40
    at arrayEach (/mnt/ssd/repositories/react-modern-calendar/node_modules/rollup-plugin-typescript2/dist/rollup-plugin-typescript2.cjs.js:534:11)
    at forEach (/mnt/ssd/repositories/react-modern-calendar/node_modules/rollup-plugin-typescript2/dist/rollup-plugin-typescript2.cjs.js:9360:14)
    at Object._onwrite (/mnt/ssd/repositories/react-modern-calendar/node_modules/rollup-plugin-typescript2/dist/rollup-plugin-typescript2.cjs.js:27429:13)
    at Object.generateBundle (/mnt/ssd/repositories/react-modern-calendar/node_modules/rollup-plugin-typescript2/dist/rollup-plugin-typescript2.cjs.js:27402:27)
    at /mnt/ssd/repositories/react-modern-calendar/node_modules/rollup/dist/shared/node-entry.js:13117:25
    at processTicksAndRejections (internal/process/task_queues.js:93:5)

```

~~The solution is to remove the `dist` after every build.~~

--- a/src/index.js
+++ b/src/index.js
@@ -574,6 +574,7 @@ function createConfig(options, entry, format, writeMeta) {
 											? undefined
 											: options.jsx || 'h',
 								},
+								files: options.entries,
 							},
 							tsconfig: options.tsconfig,
 							tsconfigOverride: {

--- a/src/index.js
+++ b/src/index.js
@@ -562,10 +562,14 @@ function createConfig(options, entry, format, writeMeta) {
 						typescript({
 							typescript: require('typescript'),
 							cacheRoot: `./node_modules/.cache/.rts2_cache_${format}`,
+							useTsconfigDeclarationDir: true,
 							tsconfigDefaults: {
 								compilerOptions: {
 									sourceMap: options.sourcemap,
 									declaration: true,
+									declarationDir: dirname(
+										pkg.types || pkg.typings || options.output,
+									),
 									jsx: 'react',
 									jsxFactory:
 										// TypeScript fails to resolve Fragments when jsxFactory

--- a/test/fixtures/ts-declaration/package.json
+++ b/test/fixtures/ts-declaration/package.json
@@ -1,0 +1,6 @@
+{
+	"name": "ts-declarations",
+	"main": "dist/index.js",
+	"types": "types/index.d.ts",
+	"source": "src/index.tsx"
+}

--- a/test/fixtures/ts-declaration/src/index.tsx
+++ b/test/fixtures/ts-declaration/src/index.tsx
@@ -1,0 +1,3 @@
+export function foo() {
+	return 42;
+}

--- a/test/fixtures/ts-declaration/tsconfig.json
+++ b/test/fixtures/ts-declaration/tsconfig.json
@@ -1,0 +1,6 @@
+{
+	"compilerOptions": {
+		"noEmit": true,
+		"allowJs": true
+	}
+}

--- a/test/fixtures/ts-declaration/types/index.d.ts
+++ b/test/fixtures/ts-declaration/types/index.d.ts
@@ -1,0 +1,1 @@
+export declare function foo(): number;

--- a/test/index.test.js
+++ b/test/index.test.js
@@ -63,13 +63,29 @@ describe('fixtures', () => {
 			expect(files.length).toMatchSnapshot();
 			// we don't realy care about the content of a sourcemap
 			files
-				.filter(file => !/\.map$/.test(file))
+				.filter(
+					file =>
+						!/\.map$/.test(file) &&
+						!fs.lstatSync(resolve(dist, file)).isDirectory(),
+				)
 				.sort(file => (/modern/.test(file) ? 1 : 0))
 				.forEach(file => {
 					expect(
 						fs.readFileSync(resolve(dist, file)).toString('utf8'),
 					).toMatchSnapshot();
 				});
+
+			// TypeScript declaration files
+			const types = resolve(`${fixturePath}/types`);
+			if (fs.existsSync(types)) {
+				const declarations = fs.readdirSync(types);
+				expect(declarations.length).toMatchSnapshot();
+				declarations.forEach(file => {
+					expect(
+						fs.readFileSync(resolve(types, file)).toString('utf8'),
+					).toMatchSnapshot();
+				});
+			}
 		},
 		TEST_TIMEOUT,
 	);

